main.nf

```diff
@@ -1745,173 +2590,277 @@ process generate_gens_data {
-	for model in !{tar}; do
-	tar -xvf $model
-	done
-    tar -xvf !{ploidy}
-	set +u
-	source activate gatk
-    export MKL_NUM_THREADS=!{task.cpus}
-    export OMP_NUM_THREADS=!{task.cpus}
-    gatk --java-options "-Xmx25g" PostprocessGermlineCNVCalls \
-        --allosomal-contig X --allosomal-contig Y \
-        --contig-ploidy-calls ploidy/!{group}-calls/ \
-        --sample-index 0 \\
-        --output-genotyped-intervals genotyped-intervals-!{group}-vs-cohort30.vcf.gz \
-        --output-genotyped-segments genotyped-segments-!{group}-vs-cohort30.vcf.gz \
-        --output-denoised-copy-ratios denoised-!{group}-vs-cohort30.vcf.gz \
-        --sequence-dictionary !{params.GENOMEDICT} \
-        --calls-shard-path !{caseshards} \
-        --model-shard-path !{modelshards}
-	'''
+	output:
+		set group, id, \
+			file("genotyped-intervals-${group}-vs-cohort30.vcf.gz"), \
+			file("genotyped-segments-${group}-vs-cohort30.vcf.gz"), \
+			file("denoised-${group}-vs-cohort30.vcf.gz") into called_gatk
+		set group, file("*versions.yml") into ch_postprocessgatk_versions
 
+	script:
+		modelshards = shard.join(' --model-shard-path ') // join each reference shard
+		caseshards = []
+		for (n = 1; n <= i.size(); n++) { // join each shard(n) that's been called
+			tmp = group+'_'+i[n-1]+'/'+group+'_'+i[n-1]+'-calls' 
+			caseshards = caseshards + tmp
+		}
+		caseshards = caseshards.join( ' --calls-shard-path ')
+ 	
+	shell:
+		version_str = postprocessgatk_version(task)
+		'''
+		THEANO_FLAGS="base_compiledir=/fs1/resources/theano"
+		for model in !{tar}; do
+		tar -xvf $model
+		done
+		tar -xvf !{ploidy}
+		set +u
+		source activate gatk
+		export MKL_NUM_THREADS=!{task.cpus}
+		export OMP_NUM_THREADS=!{task.cpus}
+		gatk --java-options "-Xmx25g" PostprocessGermlineCNVCalls \
+			--allosomal-contig X --allosomal-contig Y \
+			--contig-ploidy-calls ploidy/!{group}-calls/ \
+			--sample-index 0 \\
+			--output-genotyped-intervals genotyped-intervals-!{group}-vs-cohort30.vcf.gz \
+			--output-genotyped-segments genotyped-segments-!{group}-vs-cohort30.vcf.gz \
+			--output-denoised-copy-ratios denoised-!{group}-vs-cohort30.vcf.gz \
+			--sequence-dictionary !{params.GENOMEDICT} \
+			--calls-shard-path !{caseshards} \
+			--model-shard-path !{modelshards}
+
+		echo "!{version_str}" > "!{task.process}_versions.yml"
+		'''
+
+	stub:
+		version_str = postprocessgatk_version(task)
+		"""
+		THEANO_FLAGS="base_compiledir=/fs1/resources/theano"
+		set +u
+		source activate gatk
+		export MKL_NUM_THREADS=!{task.cpus}
+		export OMP_NUM_THREADS=!{task.cpus}
+		source activate gatk
+		touch genotyped-intervals-${group}-vs-cohort30.vcf.gz
+		touch genotyped-segments-${group}-vs-cohort30.vcf.gz
+		touch denoised-${group}-vs-cohort30.vcf.gz
+
+		echo "${version_str}" > "${task.process}_versions.yml"
+		"""
+}
+def postprocessgatk_version(task) {
+	"""${task.process}:
+	    gatk: \$(echo \$(gatk --version 2>&1) | sed 's/^.*(GATK) v//; s/ .*\$// ; s/-SNAPSHOT//')"""
 }
```

Comment:
The intent is to tab for visual indentation and space for the indentation that will end up in the out file. What do you think - OK or too ugly? (Otherwise I guess we could think about some other way to represent it).

Reply:
Also, the reason to move up the `${task.process}` is that otherwise it introduced a blank line in the final versions file.

This one looks a bit different from the others due to also being used within the shell using the `'''`. 

The result in my stub run is OK. Did you see some strange output here?

Reply:
Added a comment to these ones for clarification